crates/rooch-framework/sources/auth_validator/bitcoin_validator.move

```diff
@@ -0,0 +1,156 @@
+        let sign = vector::empty<u8>();
+        let i = 0;
+        let signature_position = ecdsa_k1_recoverable::signature_length();
+        while (i < signature_position) {
+            let value = vector::borrow(authenticator_payload, i);
+            vector::push_back(&mut sign, *value);
+            i = i + 1;
+        };
+        std::debug::print(&sign);
+        sign
+    }
+
+    /// Get the authentication key of the given authenticator from authenticator_payload.
+    public fun get_authentication_key_from_authenticator_payload(authenticator_payload: &vector<u8>): vector<u8> {
+        let public_key = get_public_key_from_authenticator_payload(authenticator_payload);
+        let addr = public_key_to_address(public_key);
+        ethereum_address::into_bytes(addr)
+    }
+
+    public fun public_key_to_address(public_key: vector<u8>): ETHAddress {
+        ethereum_address::new(public_key)
+    }
+
+    /// Get the authentication key of the given public key.
+    public fun public_key_to_authentication_key(public_key: vector<u8>): vector<u8> {
+        let addr = public_key_to_address(public_key);
+        ethereum_address::into_bytes(addr)
+    }
+
+    /// Get the authentication key option of the given account.
+    public fun get_authentication_key_option_from_account(ctx: &Context, addr: address): Option<vector<u8>> {
+        account_authentication::get_authentication_key<BitcoinValidator>(ctx, addr)
+    }
+
+    /// The authentication key exists in account or not.
+    public fun is_authentication_key_in_account(ctx: &Context, addr: address): bool {
+        option::is_some(&get_authentication_key_option_from_account(ctx, addr))
+    }
+
+    /// Extract the authentication key of the authentication key option.
+    public fun get_authentication_key_from_account(ctx: &Context, addr: address): vector<u8> {
+        option::extract(&mut get_authentication_key_option_from_account(ctx, addr))
+    }
+
+    /// Only validate the authenticator's signature.
+    public fun validate_signature(authenticator_payload: &vector<u8>, tx_hash: &vector<u8>) {
+        std::debug::print(tx_hash);
+        assert!(
+            ecdsa_k1_recoverable::verify(
+                &get_signature_from_authenticator_payload(authenticator_payload),
+                // &get_public_key_from_authenticator_payload(authenticator_payload),
+                tx_hash,
+                1
+            ),
+            auth_validator::error_invalid_authenticator()
+        );
+    }
+
+    public fun validate(ctx: &Context, authenticator_payload: vector<u8>) {
+        let tx_hash = context::tx_hash(ctx);
+        validate_signature(&authenticator_payload, &tx_hash);
+
+        // TODO compare the auth_key from lsthe payload with the auth_key from the account
+    }
+
+    fun pre_execute(
+        _ctx: &mut Context,
+    ) {}
+
+    fun post_execute(
+        ctx: &mut Context,
+    ) {
+        let account_addr = context::sender(ctx);
```

Comment:
We need to bind the address mapping here.

Reply:
yup